### ISSUE TITLE
The invoke result really confuses me

### ISSUE THREAD

**Issue by rct1985** [NeoResearch/neocompiler-eco#32]
## This is a my test smart contract (success)
``` C#
using Neo.SmartContract.Framework.Services.Neo;
namespace Neo.SmartContract
{
    public class KeyValue : Framework.SmartContract
    {
        public static byte[] BoolToByteArray(bool input){
            //return new byte[1] { input ? (byte)0 : (byte)1 };
            if(input)
                return new byte[1] { 0x01 };
            else
                return new byte[1] { 0x00 };
        }
        
        public static byte[] Main(string op, byte[] key, byte[] value)
        {
            if (op == "write") {
                Storage.Put(Storage.CurrentContext, key, value);
                //return new byte[1]{(byte)0};
                return BoolToByteArray(true);
            }
            if (op == "reads") {
                return Storage.Get(Storage.CurrentContext, key);
            }
            else{
                //return new byte[1]{(byte)1};
                return BoolToByteArray(false);
            }
        }
    }
}

``` 
After deploy it on the [neocompiler.io](https://neocompiler.io/#/ecolab), I can invoke the "write" method and the content will be written into storage, and it can be read from storage correctly.

Buy if I change the return value of "BoolToByteArray" method

## This is a my test smart contract (failed)
``` C#
using Neo.SmartContract.Framework.Services.Neo;

namespace Neo.SmartContract
{
    public class KeyValue : Framework.SmartContract
    {
        public static byte[] BoolToByteArray(bool input){
            //return new byte[1] { input ? (byte)0 : (byte)1 };
            if(input)
                return new byte[1] { 0x00 };
            else
                return new byte[1] { 0x01 };
        }
        
        public static byte[] Main(string op, byte[] key, byte[] value)
        {
            if (op == "write") {
                Storage.Put(Storage.CurrentContext, key, value);
                //return new byte[1]{(byte)0};
                return BoolToByteArray(true);
            }
            if (op == "reads") {
                return Storage.Get(Storage.CurrentContext, key);
            }
            else{
                //return new byte[1]{(byte)1};
                return BoolToByteArray(false);
            }
        }
    }
}
```
Deploy it just like the first one , I got the status of "Fault" when invoking the "write" method, and nothing was written into storage.

**Comment by igormcoelho** [comment]
That's really strange! I tried it here and the same happened to me, don't know why.... perhaps even the ternary operator is running fine, the problem is elsewhere (see this @jing).

This simple test works in both configurations (return 01, or 00, or inverted):
```
using Neo.SmartContract.Framework.Services.Neo;

namespace Neo.SmartContract
{
    public class KeyValue : Framework.SmartContract
    {
        public static byte[] BoolToByteArray(bool input){
            if(input)
                return new byte[] { 0x01 };
            else
                return new byte[] { 0x00 };
        }
        
        public static byte[] Main(string op)
        {
            bool t = op == "test";
            return BoolToByteArray(t);
        }
    }
}
```

Will need to see what is happening in opcode level, perhaps some compilation is failing...

**Comment by igormcoelho** [comment]
Oh, now it's working! The server just reset, I redeployed this contract and now it works.
So, I have a theory... perhaps you deployed this contract without paying the Storage fees (but you payed for the opposite contract), so that would explain this crazy behavior. I cannot test that, because now the server was reset (after 12-hours), but I can say that your contract works.

What we can try to do is to integrate Storage calls with RPC calls that verify if contract has set the Storage property correctly on deploy time (I think we cannot change that after deploy).

**Comment by rct1985** [comment]
## Thanks so much

**Comment by vncoelho** [comment]
closes #52 instead of 32...aheuaheua
